aptos-move/aptos-vm/src/parallel_executor/mod.rs

```diff
@@ -4,6 +4,11 @@
 mod storage_wrapper;
 mod vm_wrapper;
 
+use aptos_crypto::hash::DefaultHasher;
+use bcs::to_bytes;
+use concurrent_lru::sharded::LruCache;
```

Comment:
this lib is ok but not optimal. We can take some time to implement our own if needed.
- The number of shard is fixed (16, as of v2.0).
- The lib provides map-like cache with API like `get_or_insert(key, func-to-gen-value-if-key-absent) -> actual-value`, while we need something like "insert and return if it's a cache hit". The best we can do is to insert with a `nonce`, then claim it's a cache miss if `actual-value==nonce`. This makes false cache miss possible.

Reply:
I think I'd implement our own, should be fun and relatively small implementation where we can control most things (and testing is also not too complex). The crate is also not super-popular.